src/components/PromotionList.tsx

```diff
@@ -105,28 +105,91 @@ export default function PromotionList(props: Props): ReactElement {
 
   /**
    * If the user has not saved the promotion, save the promotion. Otherwise, delete it from their saved promotions.
+   *
+   * @param promotionId - The id of the promotion to save
    */
   const onSaveButtonClick = useCallback(
     (promotionId: string) => {
-      if (!authUser) {
-        return Promise.reject(new Error('No user is logged in.'));
-      }
       const promos = [...promotions];
       const promotion = promos.find(({ id }) => id === promotionId);
       if (promotion) {
         if (promotion.isSavedByUser) {
           promotion.isSavedByUser = false;
-          return UserService.unsavePromotion(authUser.user.id, promotionId)
+          return UserService.unsavePromotion(authUser?.user.id || '', promotionId)
             .then(() => setPromotions(promos))
             .catch(() => null);
         }
         promotion.isSavedByUser = true;
-        return UserService.savePromotion(authUser.user.id, promotionId)
+        return UserService.savePromotion(authUser?.user.id || '', promotionId)
```

Comment:
I would keep what you have above or even modify it to this: Otherwise we're going to be constructing an invalid url
```
      if (!authUser?.user?.id) {
        return Promise.reject(new Error('No user is logged in.'));
      }
```